File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int ans ;
    void com(vector<int>&nums,int idx,int temp){
        if(idx==nums.size()){
            ans+=temp;
            return;
        }
        com(nums,idx+1,temp);
        com(nums,idx+1,temp^nums[idx]);
    }
    int subsetXORSum(vector<int>& nums) {
        ans =0;
        com(nums,0,0);
        return ans;
    }
};

File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        ans = 0
        def com(idx, temp):
            nonlocal ans
            if(idx==len(nums)):
                ans += temp
                return
            com(idx+1,temp)
            com(idx+1,temp^nums[idx])
        com(0,0)
        return ans